core/src/test/scala/integration/kafka/api/AuthorizerIntegrationTest.scala

```diff
@@ -1137,8 +1136,16 @@ class AuthorizerIntegrationTest extends AbstractAuthorizerIntegrationTest {
     addAndVerifyAcls(Set(new AccessControlEntry(clientPrincipalString, WILDCARD_HOST, READ, ALLOW)), groupResource)
 
     val consumer = createConsumer()
-    consumer.subscribe(Pattern.compile(topicPattern))
-    consumer.poll(0)
+    consumer.subscribe(Pattern.compile(topicPattern), new ConsumerRebalanceListener {
+      def onPartitionsAssigned(partitions: util.Collection[TopicPartition]): Unit = {
+        assignSemaphore.release()
+      }
+      def onPartitionsRevoked(partitions: util.Collection[TopicPartition]): Unit = {
+      }})
+    TestUtils.waitUntilTrue(() => {
+      consumer.poll(Duration.ofMillis(500))
+      assignSemaphore.tryAcquire()
+    }, "Assignment did not complete on time")
```

Comment:
this is totally fine, but do we need to involve callbacks/semaphore or could we simply do the following and achieve the same:
```suggestion
    consumer.subscribe(Pattern.compile(topicPattern))
    TestUtils.waitUntilTrue(() => {
      consumer.poll(Duration.ofMillis(500))
      consumer.subscription.isEmpty
    }, "Assignment did not complete on time")
```

Reply:
In that case, yes, your suggestion would work. Personally, I like the clarity of the callbacks to drive this. The test will fail if we break the callbacks.

Reply:
Sure, both work, I'm ok with this one too